### PR TITLE
PBS job name: length maximum global.rc configurable

### LINKED ISSUE #1803: pbs 13+: job names can now have >15 characters
We should no longer truncate job names to 15 characters for PBS jobs. Do we need to make this configurable for sites stuck with PBS 11 or older?

### PR DESCRIPTION
**Pull request by matthewrmshin**
~~Plus document on how to modify a constant to bring back truncate support.~~

~~Simplest way to close #1803, but is this approach acceptable to sites stuck with an old version of PBS? If not, I'll add a global configuration for this instead.~~

Default is now the same as before, truncate at 15 characters. The length is now `global.rc` configurable.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@benfitzpatrick @arjclark please review.

File: doc/cug.tex [hunk 1]
```diff
@@ -5126,6 +5126,12 @@ \subsubsection{Default Directives Provided}
 Loadlevler, etc.) default directives are provided to set the job name
 and stdout and stderr file paths.
 
+PBS fails a job submit if the job name in \lstinline=-N name= is too long.
+For version 12 or below, this is 15 characters. For version 13, this is 236
+characters. You should modify the constant \lstinline=JOB_NAME_LEN_MAX= in the
+\lstinline=PBSHandler= class in \lstinline=lib/cylc/batch_sys_handlers/pbs.py=
```
Comment (benfitzpatrick) [inline doc/cug.tex]: Still true?

Comment (arjclark) [inline doc/cug.tex]: What, that you have to change it or that PBS just breaks at earlier versions?

File: doc/siterc.tex [hunk 2]
```diff
@@ -678,6 +678,26 @@ \subsection{[hosts]}
 \item {\em example:} see~\ref{err-tailer}
 \end{myitemize}
 
+\subparagraph[{[[[[}SYSTEM{]]]]}job name length maximum]{[hosts] \textrightarrow [[HOST]] \textrightarrow [[[batch systems]]] \textrightarrow [[[[SYSTEM]]]] \textrightarrow job name length maximum}
+
+The maximum length for job name acceptable by a batch system on a given host.
+For example, PBS 12 or older will fail a job submit if the job name has more than
```
Comment (benfitzpatrick) [inline doc/siterc.tex]: Should we mention that this only works for PBS?

File: doc/cug.tex (hunk 1 above)
Comment (benfitzpatrick) [inline doc/cug.tex]: I think this advice needs nixing completely.

**Comment by matthewrmshin** [comment]
Docs fixed.

**Comment by benfitzpatrick** [comment]
Looks good to me.

**Comment by matthewrmshin** [comment]
@arjclark please review 2.

**Comment by arjclark** [comment]
Looks OK to me. No problems from the test-battery in my environment.

**Pull request merged**